topiary-core/src/tree_sitter.rs

```diff
@@ -215,7 +215,55 @@ impl Display for LocalQueryMatch<'_> {
 // A struct to store the result of a query coverage check
 pub struct CoverageData {
     pub cover_percentage: f32,
-    pub missing_patterns: Vec<String>,
+    pub missing_patterns: Vec<LabeledSpan>,
+}
+
+impl CoverageData {
+    fn status_msg(&self) -> String {
+        match self.missing_patterns.len() {
+            0 if self.cover_percentage == 0.0 => "No queries found".into(),
+            0 => "All queries are matched".into(),
+            _ => format!("Unmatched queries: {}", self.missing_patterns.len()),
+        }
+    }
+
+    fn full_coverage(&self) -> bool {
+        self.cover_percentage == 1.0
+    }
+
+    pub fn get_result(&self) -> Result<(), FormatterError> {
+        if !self.full_coverage() {
+            return Err(FormatterError::PatternDoesNotMatch);
+        }
+        Ok(())
+    }
```

Comment:
This is intended to allow an upstream caller handling an `Ok(CoverageData { ... }` to propagate insufficient coverage as an error. Perhaps I should add this as a comment?